security-jwt/src/main/java/io/micronaut/security/token/jwt/cookie/RefreshTokenCookieConfigurationProperties.java

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.security.token.jwt.cookie;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import io.micronaut.context.annotation.ConfigurationProperties;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.http.cookie.SameSite;
+import io.micronaut.security.authentication.CookieBasedAuthenticationModeCondition;
+import io.micronaut.security.token.config.TokenConfigurationProperties;
+
+import java.time.Duration;
+import java.time.temporal.TemporalAmount;
+import java.util.Optional;
+
+/**
+ *
+ * @author Sergio del Amo
+ * @since 1.0
+ */
+@Requires(condition = CookieBasedAuthenticationModeCondition.class)
+@Requires(property = RefreshTokenCookieConfigurationProperties.PREFIX + ".enabled", notEquals = StringUtils.FALSE, defaultValue = StringUtils.TRUE)
+@ConfigurationProperties(RefreshTokenCookieConfigurationProperties.PREFIX)
+public class RefreshTokenCookieConfigurationProperties implements RefreshTokenCookieConfiguration {
+
+    public static final String PREFIX = TokenConfigurationProperties.PREFIX + ".refresh.cookie";
+
+    /**
+     * The default enable value.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static final boolean DEFAULT_ENABLED = true;
+
+    /**
+     * The default secure value.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static final boolean DEFAULT_SECURE = true;
+
+    /**
+     * The default http only value.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static final boolean DEFAULT_HTTPONLY = true;
+
+    /**
+     * The default cookie name.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static final String DEFAULT_COOKIENAME = "JWT_REFRESH_TOKEN";
+
+    /**
+     * Default Cookie Path.
+     */
+    @SuppressWarnings("WeakerAccess")
+    public static final String DEFAULT_COOKIEPATH = "/oauth/access_token";
```

Comment:
instead of a constant this should be resolved with: 

`"@Value("${" + OauthControllerConfigurationProperties.PREFIX + ".path:/oauth/access_token}") String`